instrumentation/net/http/otelhttp/transport.go

```diff
@@ -171,11 +246,15 @@ func (wb *wrappedBody) Write(p []byte) (int, error) {
 
 func (wb *wrappedBody) Read(b []byte) (int, error) {
 	n, err := wb.body.Read(b)
+	// Locally record the number of bytes read
+	wb.read.Add(int64(n))
 
 	switch err {
 	case nil:
 		// nothing to do here but fall through to the return
 	case io.EOF:
+		// Record the total number of bytes read
+		wb.record(wb.read.Load())
```

Comment:
Currently, this will only record on an EOF.  Is there a way to record on an EOF or a `close()` but only record once?

The use case I'm thinking of is if a client reads part-way and is canceled.

Reply:
Good point, I will add a test case for that and update the code.

Reply:
Addressed in 0411674549f801f9aa02f2c91a46dc6967854a03. Let me know how that looks :)

Reply:
ping @MadVikingGod